### DISCUSSION TITLE
Possible last-minute CVXPY 1.2 goals

### DISCUSSION THREAD

**Discussion by rileyjmurray** [cvxpy/cvxpy#1651] [General]
The next release of CVXPY will be version 1.2.0. We'd like to make that release this month. If you're interested in getting in some relatively simple last-minute changes, please feel free to suggest some here (you might refer to the list of open GitHub issues: https://github.com/cvxpy/cvxpy/issues). I've put suggestions from Discord below. Others should feel free to make suggestions beyond those listed here.

@akshayka @SteveDiamond @phschiele @bstellato please take a look and say which you think are most important and which you can (and/or will) work on. I'll give my 2 cents as a comment on this GH Discussion thread.

### Suggestion 1 -- DONE

*Resolved by PR #1670 and direct commits https://github.com/cvxpy/cvxpy/commit/c31e03c66aa6f77926bd7ff44621c57df38b6c4c and https://github.com/cvxpy/cvxpy/commit/40caf4f4125d8329184575d12965782244852cef. Building CVXPY is still not seamless for M1 users with this changes, so I'm leaving #1190 open.*

We should address installation issues for all supported versions of Python. This means fixing installation issues like Python 3.9 (see GitHub issues #1623 and #1604) and potentially dropping support for Python 3.6 (which is no longer supported by the Python Foundation).

Updates: #1623 is closed due to inactivity. #1604 concerns installing from pip on M1. It's unresolved and appears to be a simple result of how our ``pyproject.toml`` is set up. There's a related issue that describes a work-around for installing on M1 here: https://github.com/cvxpy/cvxpy/issues/1190#issuecomment-990418512. We could link to https://github.com/cvxpy/cvxpy/issues/1190#issuecomment-990418512 on the installation web docs before the 1.2 release. We should also try to resolve #1604 before the 1.2 release.

### Suggestion 2

address NaN propagation: https://github.com/cvxpy/cvxpy/issues/1553 and/or sending infs down to solvers https://github.com/cvxpy/cvxpy/issues/1470

### Suggestion 3

We could use Eigen's SparseCholesky module to help with DCP checking of QPs; see the later comments in https://github.com/cvxpy/cvxpy/issues/1459. 

### Suggestion 4 -- DONE (see PR 1669)

Have a flag for cvxpy to disable DPP compilation: https://github.com/cvxpy/cvxpy/issues/1340 and https://github.com/cvxpy/cvxpy/issues/1332. Per Steven's suggestion, raising a warning that DPP can be slow for parameterized problems with large parameters. The warning would say how to disable DPP https://github.com/cvxpy/cvxpy/issues/1622.

### Suggestion 5

Have MOSEK accept a keyword argument that automatically sets all  continuous problem accuracy parameters https://github.com/cvxpy/cvxpy/issues/1206

### Suggestion 6 -- DONE

Update documentation for SCS resolve behavior: https://github.com/cvxpy/cvxpy/issues/1164

### Suggestion 7

Sanity checks for solver imports: https://github.com/cvxpy/cvxpy/issues/979

**Comment by SteveDiamond** [comment]
Suggestion 4 is very important since we've promised it to many people. Happy to do this.

**Reply by rileyjmurray** [reply]
@SteveDiamond what's the status on this?

**Reply by SteveDiamond** [reply]
I made a PR https://github.com/cvxpy/cvxpy/pull/1669, happy to adjust as desired.

**Reply by rileyjmurray** [reply]
#1669 has been merged! I've updated the Discussion post and closed the appropriate issues.

**Comment by rileyjmurray** [comment]
Every suggestion except 3 and 4 could be considered bugfixes. So most suggestions don't need to be done by the end of the month to eventually get into either 1.2.x or 1.1.x. The main consideration with getting things in 1.2.0 is that it's an opportunity to make a good impression on our user base with as the first release in the 1.2.x series.

My top bugfix issues would be Suggestions 1, 7, and 2. I think Suggestion 3 would make people happy.

**Comment by SteveDiamond** [comment]
Also we need to document log_gamma and log_norm_cdf (https://github.com/cvxpy/cvxpy/issues/1610). That will be two "new" atoms in 1.2, in addition to the recently added xexp.

**Reply by rileyjmurray** [reply]
#1610 is now merged. I'm preparing a PR to resolve macOS builds. I'll update web docs while at it. I got the following errors when building the docs:

```
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/dgp/index.rst:552: WARNING: undefined label: xexp (if the link has no caption the label must precede a section header)
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/functions/index.rst:532: WARNING: undefined label: log-normcdf (if the link has no caption the label must precede a section header)
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/functions/index.rst:548: WARNING: undefined label: loggamma (if the link has no caption the label must precede a section header)
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/functions/index.rst:698: WARNING: undefined label: xexp (if the link has no caption the label must precede a section header)
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/functions/index.rst:796: WARNING: undefined label: partial-trace (if the link has no caption the label must precede a section header)
/home/riley/Documents/Research/software/cvxpydev/cvxpy/doc/source/tutorial/functions/index.rst:804: WARNING: undefined label: partial-transpose (if the link has no caption the label must precede a section header)
```
-- just leaving that as a note for myself.

**Reply by rileyjmurray** [reply]
All of the points above have been fixed.

**Comment by SteveDiamond** [comment]
Also could finish this PR https://github.com/cvxpy/cvxpy/pull/1648
